ambry-cloud/src/main/java/com.github.ambry.cloud/azure/AzureCloudDestination.java

```diff
@@ -252,6 +255,31 @@ private boolean uploadIfNotExists(BlobId blobId, long inputLength, CloudBlobMeta
     }
   }
 
+  @Override
+  public BlobReadInfo downloadBlob(BlobId blobId) throws CloudStorageException {
+    azureMetrics.blobDownloadRequestCount.inc();
+    Timer.Context storageTimer = azureMetrics.blobDownloadTime.time();
+    BlobReadInfo blobReadInfo = null;
+    try {
+      List<BlobId> blobList = new ArrayList<>();
+      blobList.add(blobId);
+      CloudBlobMetadata blobMetadata = getBlobMetadata(blobList).values().iterator().next();
+      ByteBuffer outputBuffer = ByteBuffer.allocate((int)blobMetadata.getSize());
```

Comment:
I am assuming here that blob sizes are 4-8MB, so this type conversion from long to int will not lead to any loss of data. If we support very large blobs such that their size is greater than Integer.MAX_VALUE, then this logic needs a rethink. But I am assuming that in such case, a lot of other things might need reconsideration.
@lightningrob  Please let me know if you agree with this assumption.

Reply:
Also in order to be safe, I would recommend changing the size attribute to be an integer instead of long. Will make everything cleaner. I can create a separate ticket for that.